aten/src/ATen/native/BinaryOps.cpp

```diff
@@ -194,50 +197,76 @@ Tensor& true_divide_(Tensor& self, Scalar divisor) {
   return self.div_(divisor);
 }
 
-Tensor& remainder_out(Tensor& result, const Tensor& self, const Tensor& other) {
-  auto iter = TensorIterator::binary_op(result, self, other);
-  remainder_stub(iter.device_type(), iter);
-  return result;
+
+// NOTE: integer floor division != C++ integer division. Integer division in
+//   C++ is equivalent to torch.trunc(torch.true_divide(a, b)), while
+//   floor division is equivalent to torch.floor(torch.true_divide(a, b)).
+//   For example, in Python -5 // 2 = -3, while in C++ -5 / 2 = -2.
+// NOTE: floor_divide, like Python and NumPy's floor_div(ide), participates in
+//   type promotion. However, unlike true_divide it does not always promote
+//   to a float type. This requires special dtype handling.
+Tensor& floor_divide_out(Tensor& result, const Tensor& self, const Tensor& other) {
+  // Computes the output dtype
+  auto computation_dtype = at::native::result_type(TensorList{self, other});
+
+  TORCH_CHECK(computation_dtype != ScalarType::Bool,
+    "torch.floor_divide is not implemented for scalar type bool!");
+
+  // NOTE: floordiv_stub handles the special case of integer floor division
+  if (c10::isIntegralType(computation_dtype, /*include_bool=*/false)) {
+    auto iter = TensorIterator::binary_op(result, self, other);
+    floordiv_stub(iter.device_type(), iter);
```

Comment:
nit: can you rename this floordiv_integral_stub or something?  Looking at floordiv_stub, I would have expected it implements floordiv.

Reply:
Sure.